hw/application_fpga/Makefile

```diff
@@ -381,41 +396,43 @@ application_fpga_testfw.bin: application_fpga.asc bram_fw.hex testfw.hex
 	@-$(RM) $<.tmp
 
 #-------------------------------------------------------------------
-# post-synthesis functional simulation.
+# Build testbench simulation for the design
 #-------------------------------------------------------------------
-synth_tb.vvp: $(P)/tb/tb_application_fpga.v synth.json
-	iverilog \
-		-o $@ \
-		-s tb_application_fpga synth.v $(P)/tb/tb_application_fpga.v \
+tb_application_fpga: $(SIM_VERILOG_SRCS) \
+			$(VERILOG_SRCS) \
+			$(PICORV32_SRCS) \
+			$(ICE40_SIM_CELLS) \
+			simfirmware.hex
+	python3 ./tools/app_bin_to_spram_hex.py \
+			./tb/app.bin \
+			./tb/output_spram0.hex \
+			./tb/output_spram1.hex \
+			./tb/output_spram2.hex \
+			./tb/output_spram3.hex \
+			|| { echo -e "\n -- Put your app.bin to simulate in the \"tb\" directory\n"; false; }
+	verilator \
+		-j $(shell nproc --ignore=1) \
+		--binary \
+		--cc \
+		--exe \
+		--Mdir tb_verilated \
+		--trace-fst \
+		--trace-structs \
+		--top-module tb_application_fpga_sim \
+		--timescale 1ns/1ns \
+		--timing \
+		-Wno-WIDTHEXPAND \
+		-Wno-UNOPTFLAT \
 		-DNO_ICE40_DEFAULT_ASSIGNMENTS \
-		$(ICE40_SIM_CELLS)
-	chmod -x $@
-
-synth_sim: synth_tb.vvp
-	vvp -N $<
-.PHONY: synth_sim
-
-synth_sim_vcd: synth_tb.vvp
-	vvp -N $< +vcd
-.PHONY: synth_sim_vcd
-
-#-------------------------------------------------------------------
-# post-place and route functional simulation.
-#-------------------------------------------------------------------
-route.v: application_fpga.asc  $(P)/data/$(PIN_FILE)
-	icebox_vlog -L -n application_fpga -sp  $(P)/data/$(PIN_FILE) $< > $@
-
-route_tb.vvp: route.v tb/tb_application_fpga.v
-	iverilog -o $@ -s tb_application_fpga $^ $(ICE40_SIM_CELLS)
-	chmod -x $@
-
-route_sim: route_tb.vvp
-	vvp -N $<
-.PHONY: route_sim
-
-route_sim_vcd: route_tb.vvp
-	vvp -N $< +vcd
-.PHONY: route_sim_vcd
+		-DAPP_SIZE=$(shell stat --format=%s tb/app.bin) \
```

Comment:
`-DAPP_SIZE=$(shell ls -l tb/app.bin| awk '{print $$5}') \`
For compatability with Darwin